cmd/traffic/cmd/manager/mutator/agent_injector.go

```diff
@@ -191,11 +191,16 @@ func (a *agentInjector) Inject(ctx context.Context, req *admission.AdmissionRequ
 		return nil, fmt.Errorf("invalid value %q for annotation %s", ia, agentconfig.InjectAnnotation)
 	}
 
+	config := scx.AgentConfig()
+	if agentAlreadyInjected(ctx, pod, config) {
+		dlog.Infof(ctx, "Pod %s.%s already has container %s", pod.Name, pod.Namespace, agentconfig.ContainerName)
+		return nil, nil
+	}
+
```

Comment:
Not sure what this code adds. The `addInitContainer` called later on will detect if the agent is injected already, and if it is, check if it is unchanged. That check is bypassed here.

Reply:
In that later check it will see the agent as modified. I _believe_ (may be off) that it thinks it is modified because the application container is gone and it tries to discover volumes from that container. I supplied some logs in [this comment](https://github.com/telepresenceio/telepresence/pull/3484#issuecomment-1911569638) that shows that the agent is seen as modified the second time the admission hook gets run.